lib/roby/actions/models/interface.rb

```diff
@@ -91,7 +91,7 @@ def register_action(name, action_model)
                         end
                         const_set task_model_name, task_model
                         task_model.permanent_model = self.permanent_model?
-                        action_model.returns(task_model)
+                        action_model.returns(task_model.superclass)
```

Comment:
This is a very complicated way of saying returns(Roby::Task). Which can also be said by just deleting the line...

The problem that having a default return type like this was trying to fix is to make sure that action return tasks that allow to identify them. I do start to have the same feeling than you and think that it might be just counter-productive in the end.